articles/pragmatic-documentation.md

```diff
@@ -0,0 +1,97 @@
+---
+authorName: Ognjen Blagojević
+authorGithubUsername: ognjen-blagojevic-zuehlke
+issue: 86
+title: Pragmatic documentation
+---
+# {{page.title}}
+
+Writing documentation is a task that most developers and engineers just don't
+like. The reasons for this are different, but the pattern I see is that they
+are asked to write documentation without understanding:
+* for whom the documentation is written,
+* why
+* how and
+* where?
+
+## For whom?
+  
+The first question - _Whom do I write for this documentation?_ - you should
+have a clear answer before you even start writing. Sometimes these are exactly
+certain people you know, sometimes they are existing users that you don't 
+know, and sometimes it is not yet known who the readers will be.
+
+If you are able to reach at least a certain portion of people who will read
+your documentation, do it and talk to them about the documentation you are
+writing. Explain to them that you are doing something they should use and that
+you want to do it in the best possible way. It will surely help you, because
+it is in their interest. After all, they are the ones who will eventually read
+the documentation.
+
+If your users are not yet known, it's best to try to imagine a future reader
+and try to put yourself in his/her shoes. It's not always easy, but thinking
+from the perspective of other readers -- albeit fictitious -- will help you
+decide what to write.
+
+## Why?
+
+To the second question - _Why?_ - The answer is as follows: to explain to the
+reader what he dosn't know. Remember this. If you explain to the reader what
+he already knows, your documentation will be boring and will not be read. It
+is therefore important that you understand what the reader does not know
+before starting to write the documentation. If your documentation is read by
+more than one person, the task is even more difficult. You need to know which
+things most of your readers do not know, and which only individuals don't,
+and accordingly organize the documentation.
+
+## How?
+
+When you have answers to the questions for who and why you are writing
+documents, the third and the most difficult question - _How to write it?_ -
+can be a little easier to answer. You must, the best you can, put yourself in
+the role of a person in need of help. (Because no one reads the documentation
+until he needs help.) So, put yourself in the role of reader and try to
+imagine a very concrete thing that the reader fails to do or fails to
+understand by himself. If you already have existing users, you can create
+a list of questions that they would like to have an answer. It is not good
+if the questions are general ("How is this software used?", "What is the
```

Comment:
Sometimes its nice to show "big picture" of the system. So I would avoid "not good" part here